src/popup/components/ConnectWalletForm.tsx

```diff
@@ -105,7 +146,8 @@ export const ConnectWalletForm = ({ publicKey }: ConnectWalletFormProps) => {
           required: { value: true, message: 'Wallet address URL is required.' },
           onBlur: (e: React.FocusEvent<HTMLInputElement>) => {
             getWalletCurrency(e.currentTarget.value)
-          }
+          },
+          onChange: handleOnChangeWalletAddressUrl
```

Comment:
Any reason to not do it on blur? Does it not fire right before popup closes? We can avoid debounce and stuff by doing it on blur.

Reply:
clicking outside the extension didn't trigger the onBlur